From be5a029dd8e8d678b83dc95e39f8c7c9237c8f0b Mon Sep 17 00:00:00 2001
From: Srinivas Pandruvada <srinivas.pandruvada@linux.intel.com>
Date: Fri, 20 Jun 2014 11:31:11 -0700
Subject: [PATCH] iio: accel: kxcjk1013 Fix setting fractional sampling freq

Don't check fraction part to identify register value as the values
before decimal are always unique. In this way we can can avoid
conversions and checking of fractional part.

Change-Id: Id54995aea053b0405bcff25394d70f31458405ba
Signed-off-by: Srinivas Pandruvada <srinivas.pandruvada@linux.intel.com>
---
 drivers/iio/accel/kxcjk-1013.c | 12 +++++-------
 1 file changed, 5 insertions(+), 7 deletions(-)

diff --git a/drivers/iio/accel/kxcjk-1013.c b/drivers/iio/accel/kxcjk-1013.c
index ac85006..cc1f726 100644
--- a/drivers/iio/accel/kxcjk-1013.c
+++ b/drivers/iio/accel/kxcjk-1013.c
@@ -264,26 +264,24 @@ static int kxcjk1013_chip_setup_interrupt(struct kxcjk1013_data *data,
 	return ret;
 }
 
-static int kxcjk1013_convert_freq_to_bit(int val, int val2)
+static int kxcjk1013_convert_freq_to_bit(int val)
 {
 	int i;
 
 	for (i = 0; i < ARRAY_SIZE(samp_freq_table); ++i) {
-		if (samp_freq_table[i].val == val &&
-			samp_freq_table[i].val2 == val2) {
+		if (samp_freq_table[i].val == val)
 			return samp_freq_table[i].odr_bits;
-		}
 	}
 
 	return -EINVAL;
 }
 
-static int kxcjk1013_set_odr(struct kxcjk1013_data *data, int val, int val2)
+static int kxcjk1013_set_odr(struct kxcjk1013_data *data, int val)
 {
 	int ret;
 	int odr_bits;
 
-	odr_bits = kxcjk1013_convert_freq_to_bit(val, val2 / 1000);
+	odr_bits = kxcjk1013_convert_freq_to_bit(val);
 	if (odr_bits < 0)
 		return odr_bits;
 
@@ -402,7 +400,7 @@ static int kxcjk1013_write_raw(struct iio_dev *indio_dev,
 	switch (mask) {
 	case IIO_CHAN_INFO_SAMP_FREQ:
 		mutex_lock(&data->mutex);
-		ret = kxcjk1013_set_odr(data, val, val2);
+		ret = kxcjk1013_set_odr(data, val);
 		mutex_unlock(&data->mutex);
 		break;
 	default:
-- 
1.7.11.7

